src/components/access-type.js

```diff
@@ -237,75 +256,125 @@ class AccessTypeBase extends React.Component {
+  //TODO -> need to write test cases to cover all scenarios , selectedPlan, planType , coupon, urls, story details etc.
+  initRazorPayPayment = (
+    selectedPlanObj = {},
+    planType = "",
+    storyId = "",
+    storyHeadline = "",
+    storySlug = ""
+  ) => {
     if (!selectedPlanObj) {
       console.warn("Razor pay needs a plan");
       return false;
     }
 
-    const planObject = this.makePlanObject(selectedPlanObj, planType, storyId, storyHeadline, storySlug) //we are doing this to sake of backward compatibility and will be refactored later.
+    const planObject = this.makePlanObject(
+      selectedPlanObj,
+      planType,
+      storyId,
+      storyHeadline,
+      storySlug
+    ); //we are doing this to sake of backward compatibility and will be refactored later.
     const { paymentOptions } = this.props;
-    planObject["paymentType"] = get(planObject.selectedPlan, ["recurring"]) ? "razorpay_recurring" : "razorpay";
+    planObject["paymentType"] = get(planObject.selectedPlan, ["recurring"])
+      ? "razorpay_recurring"
+      : "razorpay";
     const paymentObject = this.makePaymentObject(planObject);
     return paymentOptions.razorpay.proceed(paymentObject);
   };
 
   //TODO -> need to write test cases to cover all scenarios , selectedPlan, planType , coupon, urls, story details etc.
-  initStripePayment = (options =  {}) => { 
+  initStripePayment = (options = {}) => {
     if (!options.selectedPlan) {
       console.warn("Stripe pay needs a plan");
       return false;
     }
 
     const { paymentOptions } = this.props;
-    const paymentType = get(options.selectedPlan, ["recurring"]) ? "stripe_recurring" : "stripe";
-    const paymentObject = this.makePaymentObject({ paymentType , ...options});
-    return paymentOptions.stripe ? paymentOptions.stripe.proceed(paymentObject) : Promise.reject({message:"Payment option is loading..."})
+    const paymentType = get(options.selectedPlan, ["recurring"])
+      ? "stripe_recurring"
+      : "stripe";
+    const paymentObject = this.makePaymentObject({ paymentType, ...options });
+    return paymentOptions.stripe
+      ? paymentOptions.stripe.proceed(paymentObject)
+      : Promise.reject({ message: "Payment option is loading..." });
+  };
+
+  //TODO -> need to write test cases to cover all scenarios , selectedPlan, planType , coupon, urls, story details etc.
+  initPaypalPayment = (options = {}) => {
+    if (!options.selectedPlan) {
+      console.warn("Paypal pay needs a plan");
+      return false;
+    }
+
+    const { paymentOptions } = this.props;
+    const paymentType = get(options.selectedPlan, ["recurring"])
+      ? "paypal_recurring"
+      : "paypal";
+    const paymentObject = this.makePaymentObject({ paymentType, ...options });
+    return paymentOptions.paypal
+      ? paymentOptions.paypal
+          .proceed(paymentObject)
+          .then((response) => response.proceed(paymentObject))
```

Comment:
initStripePayment and initPaypalPayment looking almost same we can write a common fun with params?

Reply:
Right. 
Actually that's the plan that we are going to write one common function for all payments which will accept payment type param. If we do that now QA's will have to test all the payment methods again for all the publishers. That is time consuming and we have to deliver Paypal ASAP. Clients are asking about Paypal everyday therefore we are not touching any other code.